src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/TypeCast.cs

```diff
@@ -382,15 +382,14 @@ private static unsafe object CheckCastInterface_Helper(MethodTable* pTargetType,
         {
             // If object type implements IDynamicInterfaceCastable then there's one more way to check whether it implements
             // the interface.
-            if (obj.GetMethodTable()->IsIDynamicInterfaceCastable
-                && IsInstanceOfInterfaceViaIDynamicInterfaceCastable(pTargetType, obj, throwing: true))
+            // This will throw NullReferenceException if obj is null.
```

Comment:
I don't understand this comment: casting doesn't throw NullRef when obj is null. This helper is never called with null obj.

Reply:
I've inserted the comments in all places that weren't evident that it can't be null, I've changed those to Asserts to make it clearer in the source.